src/kokkos_abstraction.hpp

```diff
@@ -171,6 +177,43 @@ static struct LoopPatternTPTTRTVR {
 // Used to catch undefined behavior as it results in throwing an error
 static struct LoopPatternUndefined {
 } loop_pattern_undefined_tag;
+// Translates to a Kokkos::TeamPolicy that collapse Nteams outer loops
+// with Nthread & Nvector inner loop collapses
+template <size_t Nteam, size_t Nthread, size_t Nvector>
+struct LoopPatternCollapse {};
+
+template <typename, size_t Rank, typename T = void>
+struct LoopPatternTeam : std::false_type {};
+
+template <size_t team, size_t thread, size_t vector>
+struct LoopPatternTeam<LoopPatternCollapse<team, thread, vector>, team + thread + vector,
+                       void> : std::true_type {
+  static constexpr size_t Nvector = vector;
+  static constexpr size_t Nthread = thread;
+  static constexpr size_t Nteam = team;
+  using LoopPattern = LoopPatternCollapse<team, thread, vector>;
+};
```

Comment:
This doesn't compile in gcc <12 or nvcc < 24.5

[https://godbolt.org/z/fW56WvbKf](https://godbolt.org/z/fW56WvbKf)